packages/doc/src/MarkdownCommentStore.ts

```diff
@@ -0,0 +1,148 @@
+import { readFile } from "fs/promises";
+import { EntityDbMetadata } from "joist-codegen";
+import * as path from "path";
+import { remark } from "remark";
+import { CommentStore, EnumData, FieldSourceType } from "./CommentStore";
+import { hashFile } from "./utils";
+
+/**
+ * A Markdown CommentStore with opinionated but loose semantics around
+ * markdown document structure
+ */
+export class MarkdownCommentStore extends CommentStore {
+  private documents = new Map<string, Promise<null | ReturnType<(typeof remark)["parse"]>>>();
+
+  async forField(entity: EntityDbMetadata, fieldName: string, source: FieldSourceType, generated: boolean) {
+    return this.withGenerated(entity.name, generated, await this.findHeadingContent(entity.name, fieldName));
+  }
```

Comment:
Wdyt of making the `CommentStore` API a bit more coarse-grained, like:

```
interface CommentStore {
  getEntityComments(entity: EntityDbMetadata): EntityComments;
  getEnumComments(enum: EnumData): EnumComments;
|

interface EntityComments {
  classDoc: string;
  fieldDocs: Record<string, string>;
  sourceHash: string;
}
```

Well, I guess we want to get the `sourceHash` before going to the trouble of parsing the whole doc out. So yeah, okay, this fine-grained approach makes sense!

Reply:
If you're happy I'll leave as is for the reasons mentioned. sourceHash is the only thing we need for the best-case cached scenario, which avoids the parse and walking of the ast.

Reply:
It might be possible to follow an API like this, but retain the performance aspects

```ts
interface CommentStore {
  getEntityComments(entity: EntityDbMetadata): LazyTopic<EntityComments>;
  getEnumComments(enum: EnumData): LazyTopic<EnumComments>;
}

interface LazyTopic<T> {
  sourceHash: string;
  load(): Promise<T>
}

interface EntityComments {
  classDoc: string;
  fieldDocs: Record<string, string>;
}
```

Reply:
Another thing perhaps worth mentioning - this inverts flow which would have detrimental perf impact for large documents, Currently we request the from the CommentStore fields that are present in the source. The above would need to treat every heading in the markdown file as a possible field/property in the source, so would product a lot of needless comments that are not tied to fields in the source.

I'm err-ing on leaving as is.

Reply:
Yep, sgtm!